sycl/test/backend/cuda/primary_context.cpp

```diff
@@ -0,0 +1,82 @@
+// REQUIRES: cuda
+// RUN: %clangxx %s -I%opencl_include_dir -I%cuda_toolkit_include -o %t.out -lcuda -lsycl
+// RUN: env SYCL_DEVICE_TYPE=GPU %t.out
+
+//==---------- primary_context.cpp - SYCL cuda primary context test --------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/detail/pi_cuda.hpp>
+#include <cuda.h>
+#include <iostream>
+
+using namespace cl::sycl;
+
+class other_device_selector : public default_selector {
+public:
+
+  other_device_selector(const device &dev) : exclude_device{dev} {}
+
+  int operator()(const device &dev) const {
+    if (dev == exclude_device) {
+      return -1;
+    }
+    return default_selector::operator()(dev);
+  }
+private:
+  const device &exclude_device;
+};
+
+int main() {
+  try {
+    context c;
+  } catch (device_error e) {
+    std::cout << "Failed to create device for context" << std::endl;
+  }
+
+  device deviceA = default_selector().select_device();
```

Comment:
Please don't use default selector, it's free to return any other device. I suggest writing device selector which selects nvidia device.

Reply:
With SYCL_BE=PI_CUDA it should be alright, but you are right that it is probably safer in the long run to use a custom selector.